tools/metrics.py

```diff
@@ -75,14 +81,22 @@ def __init__(self, name, dir, output, make_flags=None):
 }
 
 
-def syscmd(*args):
+def quoted(args):
+    return " ".join(shlex.quote(word) for word in args)
+
+
+def syscmd(*args, pre_make_cmd=None):
```

Comment:
Maybe it can be just `pre_cmd`, because this `syscmd` function can be used for more than just running make (in principle).